tests/components/elmax/test_config_flow.py

```diff
@@ -26,97 +26,75 @@
 CONF_POLLING = "polling"
 
 
-def _has_error(errors):
-    return errors is not None and len(errors.keys()) > 0
-
-
-async def _bootstrap(
-    hass,
-    source=config_entries.SOURCE_USER,
-    username=MOCK_USERNAME,
-    password=MOCK_PASSWORD,
-    panel_name=MOCK_PANEL_NAME,
-    panel_pin=MOCK_PANEL_PIN,
-) -> FlowResult:
-
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
     result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": source}
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
-    if result["type"] != data_entry_flow.RESULT_TYPE_FORM or _has_error(
-        result["errors"]
-    ):
-        return result
-    result2 = await hass.config_entries.flow.async_configure(
-        result["flow_id"],
-        {
-            CONF_ELMAX_USERNAME: username,
-            CONF_ELMAX_PASSWORD: password,
-        },
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_standard_setup(hass):
+    """Test the standard setup case."""
+    # Setup once.
+    show_form_result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
-    if result2["type"] != data_entry_flow.RESULT_TYPE_FORM or _has_error(
-        result2["errors"]
+    with patch(
+        "homeassistant.components.elmax.async_setup_entry",
+        return_value=True,
     ):
-        return result2
-    result3 = await hass.config_entries.flow.async_configure(
-        result2["flow_id"],
-        {
-            CONF_ELMAX_PANEL_NAME: panel_name,
-            CONF_ELMAX_PANEL_PIN: panel_pin,
-        },
-    )
-    return result3
-
+        login_result = await hass.config_entries.flow.async_configure(
+            show_form_result["flow_id"],
+            {
+                CONF_ELMAX_USERNAME: MOCK_USERNAME,
+                CONF_ELMAX_PASSWORD: MOCK_PASSWORD,
+            },
+        )
+        result = await hass.config_entries.flow.async_configure(
+            login_result["flow_id"],
+            {
+                CONF_ELMAX_PANEL_NAME: MOCK_PANEL_NAME,
+                CONF_ELMAX_PANEL_PIN: MOCK_PANEL_PIN,
+            },
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
```

Comment:
Please block till done when patching the integration setup function to make sure no tasks are left after we exit the patch context manager.

https://github.com/home-assistant/core/blob/359affb856c186c96dbc86882ffcdc0af5f48ceb/script/scaffold/templates/config_flow/tests/test_config_flow.py#L34

Reply:
Done! Thanks.